js/piwik.js

```diff
@@ -7167,7 +7174,7 @@ if (typeof window.Piwik !== 'object') {
              * @returns number|string
              */
             this.getRememberedConsent = function () {
-                var value = getCookie('consent');
+                var value = getCookie(CONSENT_COOKIE_NAME);
```

Comment:
what if for some reason the user has a `CONSENT_REMOVED_COOKIE_NAME ` set as well? maybe we should check for this first @diosmosis ? and if `CONSENT_REMOVED_COOKIE_NAME ` is set, then we return `false`?

Reply:
This is also important for `hasRememberedConsent `

Reply:
the constant is defined above, it's not a global

Reply:
oh sorry, read that incorrectly.

Reply:
it makes sense for CONSENT_REMOVED_COOKIE_NAME to override CONSENT_COOKIE_NAME, though in this case, the user will not be tracked even if it appears they gave their consent. Though if they have both, they may never get tracked. I think in this case if CONSENT_REMOVED_COOKIE_NAME is set, we should remove CONSENT_COOKIE_NAME. This defaults to privacy and next time they will get the consent message again.

Reply:
👍 
if both are set, it is fine to remove `CONSENT_COOKIE_NAME` and assume consent is not given.

>  This defaults to privacy and next time they will get the consent message again.

This depends on the website owner. Likely the consent message would not be directly shown again.